research/blanks/model.py

```diff
@@ -194,12 +197,17 @@ def forward(self, encoder_output: torch.Tensor, model_input: torch.Tensor):
                 )
 
             for _ in range(self.n_blanks):
-                preblank_encoder_output = shift_right(preblank_encoder_output)
-                encoder_output.add_(preblank_encoder_output * abs(self.preblank_weight))
+                current_positions = shift_right(current_positions)
+                encoder_accumulator = shift_right(encoder_accumulator)
+                encoder_output.add_(encoder_accumulator)
+                encoder_accumulator = encoder_output * current_positions
+
```

Comment:
We are missing preblank weight here

Reply:
I don't think so. It first adds an accumulated change (with preblank) to output and then basically copies whatever the result was into the accumulator

Reply:
Yeah, but we don't even use the `preblank_weight` variable

Reply:
Ah, I see, sorry, didn't see that before